src/cython/sktda/preprocessing.py

```diff
@@ -0,0 +1,274 @@
+"""
+@author: Mathieu Carriere
+All rights reserved
+"""
+
+import numpy as np
+from sklearn.base          import BaseEstimator, TransformerMixin
+from sklearn.preprocessing import StandardScaler
+
+#############################################
+# Preprocessing #############################
+#############################################
+
+class BirthPersistenceTransform(BaseEstimator, TransformerMixin):
+    """
+    This is a class for the affine transformation (x,y) -> (x,y-x) to be applied on persistence diagrams. It is a particular scaler for persistence diagram that can be given as input for the DiagramPreprocessor class.
+    """
+    def __init__(self):
+        """
+        Constructor for BirthPersistenceTransform class.
+        """
+        return None
+
+    def fit(self, X, y=None):
+        """
+        Fit the BirthPersistenceTransform class on a list of persistence diagrams (this function actually does nothing but is useful when BirthPersistenceTransform is included in a scikit-learn Pipeline).
+
+        Parameters:
+            X (list of n x 2 or n x 1 numpy arrays): input persistence diagrams.
+            y (n x 1 array): persistence diagram labels (unused).
+        """
+        return self
+
+    def transform(self, X):
+        """
+        Apply the BirthPersistenceTransform function on the persistence diagrams.
+
+        Parameters:
+            X (list of n x 2 or n x 1 numpy arrays): input persistence diagrams.
+
+        Returns:
+            Xfit (list of n x 2 numpy arrays): transformed persistence diagrams.
+        """
+        Xfit = np.matmul(X, np.array([[1., -1.],[0., 1.]]))
+        return Xfit
+
+
+class DiagramPreprocessor(BaseEstimator, TransformerMixin):
```

Comment:
The name looks awfully generic, preprocessing diagrams is what all the classes in this file do. Would DiagramScaler make sense? Or maybe not since it calls scalers instead of being one itself?
Is it standard with sklearn to have a wrapper for a list of scalers instead of using the scalers directly?

Reply:
I agree, `DiagramScaler` is better. I do not think this kind of wrapper is very standard in `sklearn`, but I did not want to re-implement all the traditional scalers of `sklearn` for persistence diagrams...